src/cpp/src/llm_pipeline_static.cpp

```diff
@@ -698,22 +706,21 @@ EncodedResults StaticLLMPipeline::generate(
     // NB: Copy KV-cache tensors from prefill model to kvcache model
     const auto& kvcache_compiled = m_kvcache_request.get_compiled_model();
     for (int i = 0; i < kvcache_compiled.outputs().size() - 1; ++i) {
-
-        const auto& output_name = kvcache_compiled.outputs()[kStartOutputKVCacheLayers + i].get_any_name();
-        auto prefill_out_tensor = m_prefill_request.get_tensor(output_name);
-        auto prefill_out_slice = make_tensor_slice(
-            prefill_out_tensor, m_kvcache_desc.dim, m_kvcache_desc.max_prompt_size - m_kvcache_desc.num_stored_tokens, m_kvcache_desc.max_prompt_size
-        );
-
         const auto& input_name = kvcache_compiled.inputs()[kStartInputKVCacheLayers + i].get_any_name();
-        auto kvcache_in_tensor = m_kvcache_request.get_tensor(input_name);
+        auto kvcache_in_tensor = m_kvcache_request.get_tensor(input_name); 
         fill_tensor<ov::float16>(kvcache_in_tensor, 0);
 
-        auto kvcache_in_slice = make_tensor_slice(
-            kvcache_in_tensor, m_kvcache_desc.dim, 0u, m_kvcache_desc.num_stored_tokens
-        );
+        const auto& output_name = kvcache_compiled.outputs()[kStartOutputKVCacheLayers + i].get_any_name();
+        auto prefill_out_tensor = m_prefill_request.get_tensor(output_name);
 
-        prefill_out_slice.copy_to(kvcache_in_slice);
+        uint16_t* src_ptr = (uint16_t*)prefill_out_tensor.data() + (m_kvcache_desc.max_prompt_size - m_kvcache_desc.num_stored_tokens) * get_data_size(prefill_out_tensor.get_shape(), m_kvcache_desc.dim);
+        uint16_t* dst_ptr = (uint16_t*)kvcache_in_tensor.data();
+        int src_gap_size = get_data_size(prefill_out_tensor.get_shape(), m_kvcache_desc.dim - 1);
+        int dst_gap_size = get_data_size(kvcache_in_tensor.get_shape(), m_kvcache_desc.dim - 1);
+        int copy_size = get_data_size(prefill_out_tensor.get_shape(), m_kvcache_desc.dim);
+        for(int k = 0; k < (m_kvcache_desc.dim > 0 ? kvcache_in_tensor.get_shape().at(m_kvcache_desc.dim - 1) : 1); k++) {
+            memcpy(dst_ptr + k * dst_gap_size, src_ptr + k * src_gap_size, copy_size * sizeof(ov::float16) * m_kvcache_desc.num_stored_tokens);
```

Comment:
1. It might make sense to make all calculation right in the bytes from the start and don't multiply on data type here.
2. It seems that logic is correct for `m_kvcache_desc.dim == 1` and might be `m_kvcache_desc.dim == 0` (if such case exists). However it seems that we won't cover full tensor for `m_kvcache_desc.dim == 2`, as we will go through the `shape[1] * get_step_for(..., 2)` here (k=0..shape[1]), what is only step for `dim == 0` and not the full tensor size. What do you think?

Reply:
Maybe I don't fully understand what you mean in point 2, when we go through the shape[1] * get_step_for(..., 1) here (k=0..shape[1], dst_gap_size=get_step_for(..., 1), why it is only step for `dim == 0`?